components/scrubber/scrubber.go

```diff
@@ -55,33 +61,61 @@ type Scrub interface {
 var Default Scrubber = newScrubberImpl()
 
 func newScrubberImpl() *scrubberImpl {
-	fieldIndex := make(map[string]Sanitisatiser, len(HashedFieldNames)+len(RedactedFieldNames))
+	hashedRegex, err := regexp.Compile("(?i)(" + strings.Join(HashedFieldNames, "|") + ")")
+	if err != nil {
+		panic(fmt.Errorf("cannot compile hashed regex: %w", err))
+	}
+
+	redactedRegex, err := regexp.Compile("(?i)(" + strings.Join(RedactedFieldNames, "|") + ")")
+	if err != nil {
+		panic(fmt.Errorf("cannot compile redacted regex: %w", err))
+	}
+
+	fieldIndex := make(map[string]*fieldEntry, len(HashedFieldNames)+len(RedactedFieldNames)+2)
 	for _, f := range HashedFieldNames {
-		fieldIndex[strings.ToLower(f)] = SanitiseHash
+		fieldIndex[f] = &fieldEntry{sanitisatiser: SanitiseHash}
 	}
+	fieldIndex["workspaceId"] = &fieldEntry{sanitisatiser: SanitiseHash}
+	fieldIndex["WorkspaceId"] = &fieldEntry{sanitisatiser: SanitiseHash}
 	for _, f := range RedactedFieldNames {
-		fieldIndex[strings.ToLower(f)] = SanitiseRedact
+		fieldIndex[f] = &fieldEntry{sanitisatiser: SanitiseRedact}
 	}
+
 	res := &scrubberImpl{
+		RegexpIndex: map[*regexp.Regexp]Sanitisatiser{
+			hashedRegex:   SanitiseHash,
+			redactedRegex: SanitiseRedact,
+		},
 		FieldIndex: fieldIndex,
+		MaxSize:    1024 * 1024, // 1Mb
 	}
 	res.Walker = &structScrubber{Parent: res}
+
 	return res
 }
 
+type fieldEntry struct {
+	sanitisatiser Sanitisatiser
+	frequency     int64
```

Comment:
The use of `frequency` here looks a lot like an LRU. We use https://github.com/hashicorp/golang-lru for LRU caches in multiple places already. How suitable would that be here?

Reply:
lru just kick out some? not less frequently used, I think we will need to use lfu

I tried adding some library like https://dgraph.io/blog/post/introducing-ristretto-high-perf-go-cache/ but it makes generally scrubbing slower.

I think here it is a rather safety guard, not something what is going to happen a lot with proper usage of logging to compromise performance with libraries.